src/EPR.SubsidiaryBulkUpload.Application/Services/CompaniesHouseDataProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿using EPR.SubsidiaryBulkUpload.Application.Models;
+using EPR.SubsidiaryBulkUpload.Application.Services.Interfaces;
+
+namespace EPR.SubsidiaryBulkUpload.Application.Services;
+
+public class CompaniesHouseDataProvider(ICompaniesHouseLookupService companiesHouseLookupService, ISubsidiaryService subsidiaryService) : ICompaniesHouseDataProvider
+{
+    private readonly ICompaniesHouseLookupService companiesHouseLookupService = companiesHouseLookupService;
+    private readonly ISubsidiaryService subsidiaryService = subsidiaryService;
+
+    public async Task<bool> SetCompaniesHouseData(OrganisationModel subsidiaryModel)
+    {
+        var dataRetrieved = false;
+
+        // Try get locally...
+        var response = await subsidiaryService.GetCompanyByOrgIdFromTableStorage(subsidiaryModel.CompaniesHouseNumber);
+
+        if (response != null)
+        {
+            dataRetrieved = true;
+            subsidiaryModel.Address = response.Address;
+        }
+        else
+        {
+            // Try get remotely through CH API
```

Comment:
Not including any commented lines?

Reply:
there are few more comments in other classes. there are really helpful for readability.